lib/dl_connector_mysql/dl_connector_mysql/core/adapters_mysql.py

```diff
@@ -17,7 +18,39 @@ class MySQLAdapter(BaseMySQLAdapter, BaseClassicAdapter[MySQLConnTargetDTO]):
     execution_options = {
         "stream_results": True,
     }
-    _error_transformer: ClassVar[DbErrorTransformer] = sync_mysql_db_error_transformer
+    _error_transformer: typing.ClassVar[DbErrorTransformer] = sync_mysql_db_error_transformer
 
     def get_connect_args(self) -> dict:
-        return dict(charset="utf8", local_infile=0, ssl={"ssl_check_hostname": False})
+        connect_args = dict(
+            super().get_connect_args(),
+            charset="utf8",
+            local_infile=0,
+        )
+
+        if self._target_dto.ssl_enable:
+            connect_args["ssl"] = (
```

Comment:
Potentially we can avoid using ca_file and use SSLContext, and use ca_data there to avoid creating temp ca files:
https://github.com/PyMySQL/PyMySQL/blob/main/pymysql/connections.py#L133